src/aspire/estimation/covar2d.py

```diff
@@ -166,43 +168,49 @@ def identity(x):
         if mean_coeff is None:
             mean_coeff = self.get_mean(coeffs, ctf_fb, ctf_idx)
 
-        block_partition = blk_diag_partition(ctf_fb[0])
-        b_coeff = blk_diag_zeros(block_partition, dtype=coeffs.dtype)
-        b_noise = blk_diag_zeros(block_partition, dtype=coeffs.dtype)
+        block_partition = get_partition(ctf_fb[0])
+        b_coeff = BlockDiagonal.zeros(block_partition, dtype=coeffs.dtype)
+        b_noise = BlockDiagonal.zeros(block_partition, dtype=coeffs.dtype)
         A = []
         for k in range(0, len(ctf_fb)):
-            A.append(blk_diag_zeros(block_partition, dtype=coeffs.dtype))
+            A.append(BlockDiagonal.zeros(block_partition, dtype=coeffs.dtype))
 
-        M = blk_diag_zeros(block_partition, dtype=coeffs.dtype)
+        M = BlockDiagonal.zeros(block_partition, dtype=coeffs.dtype)
 
         for k in np.unique(ctf_idx[:]):
 
             coeff_k = coeffs[:, ctf_idx == k]
             weight = np.size(coeff_k, 1)/np.size(coeffs, 1)
 
             ctf_fb_k = ctf_fb[k]
-            ctf_fb_k_t = blk_diag_transpose(ctf_fb_k)
-            mean_coeff_k = blk_diag_apply(ctf_fb_k, mean_coeff)
+            ctf_fb_k_t = ctf_fb_k.T
+            mean_coeff_k = ctf_fb_k.apply(mean_coeff)
             covar_coeff_k = self._get_covar(coeff_k, mean_coeff_k)
 
-            b_coeff = blk_diag_add(b_coeff, blk_diag_mult(ctf_fb_k_t,
-                blk_diag_mult(covar_coeff_k, blk_diag_mult(ctf_fb_k, weight))))
+            b_coeff += ctf_fb_k_t @ covar_coeff_k @ (ctf_fb_k * weight)
 
-            A_temp = blk_diag_mult(ctf_fb_k_t, ctf_fb_k)
-            b_noise = blk_diag_add(b_noise, blk_diag_mult(A_temp, weight))
+            A_temp = ctf_fb_k_t @ ctf_fb_k
+            b_noise += A_temp * weight
 
-            A[k] = blk_diag_mult(A_temp, np.sqrt(weight))
-            M = blk_diag_add(M, A[k])
+            A[k] = A_temp * np.sqrt(weight)
+            M += A[k]
 
         if covar_est_opt['shrinker'] == 'None':
-            b = blk_diag_add(b_coeff, blk_diag_mult(-noise_var, b_noise))
+            # XXXX todo fix rmul
```

Comment:
Please remove unnecessary statements that are commented out.

Reply:
sure, todo in these cases mean I will be addressing them before the request is finalized.